webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -1,23 +1,24 @@
 <template>
-  <ds-button v-if="totalNotifications <= 0" class="notifications-menu" disabled icon="bell">
-    {{ totalNotifications }}
+  <ds-button v-if="!notificationsCount" class="notifications-menu" disabled icon="bell">
+    {{ unreadNotificationsCount }}
   </ds-button>
   <dropdown v-else class="notifications-menu" :placement="placement">
     <template slot="default" slot-scope="{ toggleMenu }">
-      <ds-button primary icon="bell" @click.prevent="toggleMenu">
-        {{ totalNotifications }}
+      <ds-button :primary="!!unreadNotificationsCount" icon="bell" @click.prevent="toggleMenu">
+        {{ unreadNotificationsCount }}
       </ds-button>
     </template>
     <template slot="popover">
       <div class="notifications-menu-popover">
-        <notification-list :notifications="notifications" @markAsRead="markAsRead" />
+        <notification-list :notifications="displayedNotifications" @markAsRead="markAsRead" />
       </div>
     </template>
   </dropdown>
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-26T06:48:08Z" title="Thursday, September 26th 2019, 8:48:08 am +02:00">Sep 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,23 +1,35 @@
 <template>
-  <ds-button v-if="totalNotifications <= 0" class="notifications-menu" disabled icon="bell">
-    {{ totalNotifications }}
-  </ds-button>
+  <div
+    v-if="totalNotifications <= 0"
+    class="notifications-menu-pointer"
+    @click.prevent="updateNotifications"
+  >
+    <ds-button class="notifications-menu" disabled icon="bell">{{ unreadNotifications }}</ds-button>
+  </div>
   <dropdown v-else class="notifications-menu" :placement="placement">
     <template slot="default" slot-scope="{ toggleMenu }">
-      <ds-button primary icon="bell" @click.prevent="toggleMenu">
-        {{ totalNotifications }}
+      <ds-button
+        primary
+        icon="bell"
+        @click.prevent="
+          toggleMenu()
+          updateNotifications()
```


~This looks odd, maybe you can make a method to wrap those two?~

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-26T07:20:28Z" title="Thursday, September 26th 2019, 9:20:28 am +02:00">Sep 26, 2019</time>_

---


Actually I prefer not to `updateNotifications` when i click on the notifications. It could happen that the list gets updated just when I want to click on a notification.

Can't we just let the notifications update on their own?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-26T07:41:16Z" title="Thursday, September 26th 2019, 9:41:16 am +02:00">Sep 26, 2019</time>_

---


@roschaefer it might not be clear, but clicking on it to update only works if there are no `totalNotifications`, so it's not possible to update if you want to click on a notification

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-26T07:45:27Z" title="Thursday, September 26th 2019, 9:45:27 am +02:00">Sep 26, 2019</time>_

---


If I need to have a quick conversation with somebody this way I am able to get the notifications menu as quick updated as possible. Also in case the automatic update stops for any reason.

I discussed this with @datenbrei and we agreed to do it this way.

To have the menu not updated while it is open we could say:
```
 @click.prevent="
          await updateNotifications()
          toggleMenu()
 "
```
But then there will be a delay before the menu opens …
I my eyes there is no big possible harm if the open menu sometimes updates after I click on it.

PS: The `toggleMenu` function is undefined in any method as far I have tried …

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-26T12:54:36Z" title="Thursday, September 26th 2019, 2:54:36 pm +02:00">Sep 26, 2019</time>_

---


I think my head explodes right now.

Up to you. If you feel it's a great idea to update the list when you click on notifications - feel free to do it that way. I would prefer to implement the smallest solution first and get some user feedback before moving on the more sophisticated solution. But up to you. I have confidence in your decisions :wink: